.github/workflows/model_uploader.yml

```diff
@@ -0,0 +1,321 @@
+              ${{ github.event.inputs.model_id }} ${{ github.event.inputs.model_version }} TORCH_SCRIPT)
+          aws s3api head-object --bucket opensearch-exp --key $TORCH_FILE_PATH > /dev/null 2>&1 || TORCH_MODEL_NOT_EXIST=true
+          if [[ -z $TORCH_MODEL_NOT_EXIST ]];
+          then
+            echo "TORCH_SCRIPT Model already exists on model hub."
+            exit 1
+          fi
+      - name: Check if ONNX Model Exists
+        if: github.event.inputs.tracing_format == 'ONNX' || github.event.inputs.tracing_format == 'BOTH'
+        run: |
+          ONNX_FILE_PATH=$(python utils/model_uploader/save_model_file_path_to_env.py \
+            ${{ github.event.inputs.model_id }} ${{ github.event.inputs.model_version }} ONNX)
+          aws s3api head-object --bucket opensearch-exp --key $ONNX_FILE_PATH > /dev/null 2>&1 || ONNX_MODEL_NOT_EXIST=true
+          if [[ -z $ONNX_MODEL_NOT_EXIST ]];
+          then
+            echo "TORCH_SCRIPT Model already exists on model hub."
+            exit 1;
+          fi
+  
+  # Step 3: Trace the model, Verify the embeddings & Upload the model files as artifacts
+  model-auto-tracing:
+    needs: checking-out-model-hub
+    name: model-auto-tracing
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      matrix:
+        cluster: ["opensearch"]
+        secured: ["true"]
+        entry:
+          - { opensearch_version: 2.7.0 }
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Export Arguments
+        run: | 
+          echo "MODEL_ID=${{ github.event.inputs.model_id }}" >> $GITHUB_ENV
+          echo "MODEL_VERSION=${{ github.event.inputs.model_version }}" >> $GITHUB_ENV
+          echo "TRACING_FORMAT=${{ github.event.inputs.tracing_format }}" >> $GITHUB_ENV
+          echo "EMBEDDING_DIMENSION=${{ github.event.inputs.embedding_dimension }}" >> $GITHUB_ENV
+          echo "POOLING_MODE=${{ github.event.inputs.pooling_mode }}" >> $GITHUB_ENV     
+      - name: Autotracing ${{ matrix.cluster }} secured=${{ matrix.secured }} version=${{matrix.entry.opensearch_version}}
+        run: "./.ci/run-tests ${{ matrix.cluster }} ${{ matrix.secured }} ${{ matrix.entry.opensearch_version }} trace"
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with: 
+          name: upload
+          path: ./upload/
+          retention-days: 5
+          if-no-files-found: error
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ secrets.MODEL_UPLOADER_AWS_REGION }}
+          role-to-assume: ${{ secrets.MODEL_UPLOADER_ROLE }}
+          role-session-name: model-auto-tracing
+      - name: Dryrun model uploading
+        id: dryrun_model_uploading
+        run: |
+          aws s3 sync ./upload/ s3://opensearch-exp/ml-models/huggingface/sentence-transformers/ --dryrun
```

Comment:
1. `opensearch-exp` don't we need to replace this with `opensearch-models`
2. I was thinking `huggingface/sentence-transformers` will be set from the model id we are providing to keep the flow more generic? Or at least for now let's keep `sentence-transformers` dynamic. For example: if we give model id as `sentence-transformers/msmarco-distilbert-base-tas-b` then it will create a file path for : 
`opensearch-models/ml-models/huggingface/sentence-transformers/msmarco-distilbert-base-tas-b`. And for `meta-llama/Llama-2-7b-chat-hf` it'll create file path: `opensearch-models/ml-models/huggingface/meta-llama/Llama-2-7b-chat-hf`.

Reply:
Will make those changes!